galsim/optics.py

```diff
@@ -174,8 +200,11 @@ class OpticalPSF(GSObject):
     """
 
     # Initialization parameters of the object, with type information
-    _req_params = { "lam_over_diam" : float }
+    _req_params = { }
     _opt_params = {
+        "lam_over_diam" : float ,
+        "lam" : float ,
```

Comment:
This will work, but I think it may give slightly better error messages if `lam_over_diam` and `lam` are considered a "single" item.  One and only one of them is required.  

It's not quite as nice as the other examples of this because of the fact that `diam` has to accompany `lam`.  Also, your runtime checks here do much the same job as the config checks.  But since these parameters are conceptually closer to the meaning of `_single_params` than to `opt_params`, it seems worth moving them:
```
_single_params = [ { "lam_over_diam" : float , "lam" : float } ]
```